File: MidTier/MicroService/buyerms/src/main/java/com/emart/buyer/service/CartService.java
package com.emart.buyer.service;

import java.util.List;

import com.emart.buyer.model.CartModel;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Service;

public interface CartService {
    /**
     * Get cart
     * @param userId
     * @return List<CartModel>
     */
    public List<CartModel> getCart(Integer userId);

    /**
     * Add item to buyer's cart.
     * @param model CartModel
     * @return the number of items in buyer's cart
     */
    public Integer add(CartModel model);

    /**
     * Delete item from buyer's cart.
     * @param model CartModel
     * @return the number of items in buyer's cart
     */
    public Integer delete(CartModel model);

}

File: FrontEnt/src/app/services/cart.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpParams,HttpHeaders} from '@angular/common/http';

const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};
@Injectable({
  providedIn: 'root'
})
export class CartService {
  items = [];

  constructor(private http:HttpClient)  {}

   /* add to cart */
  addToCart(item){
    this.items.push(item);
      return this.http.post(`buyer/cart/add`, JSON.stringify(item), httpOptions);
  }

  /* search by userId */
  getCart(userId: string) {
    return this.http.get(`buyer/cart`,
      { params: new HttpParams()
                .set('userId', userId)});
  }

    /* delete item by userId */
    deteleitem(id: string) {
      return this.http.delete(`buyer/cart/delete`,
        { params: new HttpParams()
                  .set('Id', id)});
    }
  
  clearCart(){
    this.items = [];
    return this.items;
  }

}